vespa_analysis/vespa_eval.py

```diff
@@ -0,0 +1,159 @@
+            gen_aln = msa_dict[gen_id]
+            cons_class = cons_dict[ref_id]
+            vespa_map_out = cwd+'/vespa_scores/'+gen_id+'_vespa_scores.csv'
+            cmd.write('python vespa_msa.py --vespa_map_out '+vespa_map_out+' --ref_id '+ref_id+' --ref_aln '+ref_aln.replace('-','_')+' --cons_class '+cons_class+' --gen_id '+gen_id+' --gen_aln "'+gen_aln.replace('-','_')+'"\n')
+    os.system('parallel -j '+str(threads)+' < vespa_msa_cmd.list')   
+    os.system('cat vespa_msa/cons_stats/*.csv > vespa_msa/'+gen_name+'_cons_scores.csv')
+    os.system('rm -r vespa_msa/cons_stats/')
+    os.system('cat vespa_msa/sav_stats/*.csv > vespa_msa/'+gen_name+'_sav_scores.csv')
+    os.system('rm -r vespa_msa/sav_stats/')
+
+parser = argparse.ArgumentParser(description='Calculates SAV scores using VESPA')
+parser.add_argument('-r', type=str, action='store', help='FULL Path Fasta file containing the Reference sequences',required=True)
+parser.add_argument('-g', type=str, action='store', help='FULL Path Fasta file containing the Generated sequences',required=True)
+parser.add_argument('--cache', type=str, action='store', help='FULL Path of caching folder to store Prott5 weights',required=True)
+parser.add_argument('--threads', type=str, action='store', help='Number of threads to be used in parallelization',required=True)
+args = parser.parse_args()
+
+## Parsing Input ##
+ref_fasta = args.r
+ref_name = ref_fasta.split('/')[-1]
+ref_name = ref_name.split('.')[0]
+gen_fasta = args.g 
+gen_name = gen_fasta.split('/')[-1]
+gen_name = gen_name.split('.')[0]
+cache_path = args.cache
+threads = int(args.threads)
+cwd = os.getcwd()
+
+ref_dict = fasta2dict(ref_fasta)
+gen_dict = fasta2dict(gen_fasta)
+
+dummy_thread = 0
+if len(gen_dict)%threads != 0:
+    dummy_thread = threads
+    for i in range(threads):
+        dummy_thread = dummy_thread-1
+        if len(gen_dict)%dummy_thread ==0:
+            threads = dummy_thread
+            break
+
+## Running VESPA and MMSEQ ##
+vespa_out = run_vespa(ref_fasta,cache_path)
+mmseq_out = run_mmseq(gen_fasta,ref_fasta,gen_name,threads)
+
+## Mapping VESPA scores to Mutants ##
+pair_map (gen_dict,ref_dict,mmseq_out,vespa_out,threads)
+
+## Computing SAV and Conservation statistics ##
+msa_map (ref_fasta,gen_fasta,threads)
+
+## Plotting ##
+df_sav = pd.read_csv('vespa_msa/'+gen_name+'_sav_scores.csv',header=None)
+df_sav_avg = df_sav.mean()
+sav_avg_arr = np.array(list(df_sav_avg.values))
+df_cons = pd.read_csv('vespa_msa/'+gen_name+'_cons_scores.csv',header=None)
+df_cons_avg = df_cons.mean()
+cons_avg_arr = np.array(list(df_cons_avg.values))
+spearman_corr = scipy.stats.mstats.spearmanr(cons_avg_arr,sav_avg_arr)
+
+df_all_avg = pd.DataFrame()
+df_all_avg['Average Mutant SAV Scores'] = list(df_sav_avg.values)
+df_all_avg['Average Wildtype Conservation Class'] = list(df_cons_avg.values)
+
+## Plotting Positional SAV Effect and Conservation Classes ##
+plt.figure(figsize=(15,8))
```

Comment:
it will be better to group all the plotting stuff into one function then call it with the necessary inputs to save the figure, what do you think?